### PR TITLE
Merge tests which verify floating points registers save/restore during context switch

### PR DESCRIPTION
**Pull request by cjhackillinois**
I added two units tests which utilize FP registers and context switch. This should allow code to use the GP registers or FP registers without an issue. Note, this code tests the use of the floating point registers (lower 64-bits) and not SIMD vector registers (upper 64-bits).

### CODE REVIEW

File: include/fastcontext/arm-ucontext.h
Comment (janciesko) [inline include/fastcontext/arm-ucontext.h:1]: Unrelated changes

File: src/fastcontext/asm.S
Comment (janciesko) [inline src/fastcontext/asm.S:1]: Unrelated changes

**Review by janciesko — CHANGES_REQUESTED** [review]

File: src/fastcontext/asm.S
Comment (cjhackillinois) [reply src/fastcontext/asm.S:1]: I'll look at remove those changes next week.

**Comment by janciesko** [comment]
Any updates on this?

File: include/fastcontext/arm-ucontext.h
Comment (cjhackillinois) [reply include/fastcontext/arm-ucontext.h:1]: can you point to the unrelated changes?

File: src/fastcontext/asm.S
Comment (cjhackillinois) [reply src/fastcontext/asm.S:1]: can you point to the unrelated changes?

**Comment by cjhackillinois** [comment]
I switched laptops and needed to get my environment back up and working. Everything seems to work with `-O3` and fast context.

**Comment by cjhackillinois** [comment]
@janciesko still learning github here. I look at my repo and it wants me to discard 7 commits before I can sync. 

image

I assume I would lose my 2 new unit tests, so for now I'm now doing anything. What do you recommend?

**Comment by janciesko** [comment]
Closing in favor of https://github.com/Qthreads/qthreads/pull/135

**Pull request closed without merging**